src/scaffold.test.ts

```diff
@@ -0,0 +1,122 @@
+import * as assert from "assert";
+import * as sinon from "sinon";
+import * as vscode from "vscode";
+import {
+  ScaffoldV1TemplateListApiVersionEnum,
+  ScaffoldV1TemplateListDataInner,
+  ScaffoldV1TemplateListDataInnerApiVersionEnum,
+  ScaffoldV1TemplateListDataInnerKindEnum,
+  ScaffoldV1TemplateListKindEnum,
+} from "./clients/scaffoldingService";
+import * as scaffold from "./scaffold";
+
+describe("scaffoldProjectRequest", () => {
+  let sandbox: sinon.SinonSandbox;
+  let getTemplatesStub: sinon.SinonStub;
+  let quickPickStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    sandbox = sinon.createSandbox();
+
+    const templateData: ScaffoldV1TemplateListDataInner[] = [
+      {
+        api_version: ScaffoldV1TemplateListDataInnerApiVersionEnum.ScaffoldV1,
+        kind: ScaffoldV1TemplateListDataInnerKindEnum.Template,
+        metadata: { self: undefined },
+        spec: {
+          name: "kafka-js",
+          display_name: "Kafka Client In JavaScript",
+          description: "A simple JavaScript project",
+          tags: ["producer", "consumer", "javascript"],
+        },
+      },
+      {
+        api_version: ScaffoldV1TemplateListDataInnerApiVersionEnum.ScaffoldV1,
+        kind: ScaffoldV1TemplateListDataInnerKindEnum.Template,
+        metadata: { self: undefined },
+        spec: {
+          name: "flink-sql",
+          display_name: "Flink SQL Application",
+          description: "A Flink SQL application",
+          tags: ["apache flink", "table api"],
+        },
+      },
+      {
+        api_version: ScaffoldV1TemplateListDataInnerApiVersionEnum.ScaffoldV1,
+        kind: ScaffoldV1TemplateListDataInnerKindEnum.Template,
+        metadata: { self: undefined },
+        spec: {
+          name: "other-template",
+          display_name: "Other Template",
+          description: "Another template",
+          tags: ["other"],
+        },
+      },
+    ];
+
+    const mockTemplates = {
+      api_version: ScaffoldV1TemplateListApiVersionEnum.ScaffoldV1,
+      kind: ScaffoldV1TemplateListKindEnum.TemplateList,
+      metadata: {
+        first: true,
+        last: true,
+        total_size: 3,
+      },
+      data: new Set(templateData),
+    };
+
+    getTemplatesStub = sandbox.stub(scaffold, "getTemplatesList").resolves(mockTemplates);
```

Comment:
We have a bit of a problem here: because of the fact that `getTemplatesList()` is in the same module as `scaffoldProjectRequest`, this stub is never actually applied properly. Adding an info log immediately after the `getTemplatesList()` call proves we're still making a real request to the scaffolding service:
<img width="1970" alt="image" src="https://github.com/user-attachments/assets/69fbf4eb-be7b-477f-81c6-99bbb68edd71" />

Here's a minimal example of that same stub issue behavior:
![image](https://github.com/user-attachments/assets/96feb57f-4bba-4854-b890-c9407b120db9)

So we have a decision to make -- do we move `getTemplatesList()`out of `scaffold.ts` so it can be properly stubbed, or do we start building up a class (whose methods can all be stubbed)?

cc @noeldevelops

Reply:
hmmmm should we start in on a class? with an eye on increasing test coverage of these items

Reply:
Another option is moving `resourceScaffoldProjectRequest` and `scaffoldProjectRequest` into a new `src/commands/projectGeneration.ts` (since they're already registered as command-functions) and leave all of the existing `scaffold.ts` as-is